fn.go

```diff
@@ -139,7 +139,7 @@ func (x Fn) SizeHint() int { return 32 }
 // Marshal implements the surge.Marshaler interface.
 func (x Fn) Marshal(w io.Writer, m int) (int, error) {
```

Comment:
If the intent is to implement the new `surge` interfaces, then you need to change this function signature to `Marshal(buf []byte, rem int) ([]byte, int, error)`, where the returned `[]byte` slice is the tail (unconsumed) part of the buffer.

Reply:
The marshalling for this and also `Fp` were implemented before considering the new surge, so only the `Point` type implements the new interface. There is an issue to update the implementations for `Fn` and `Fp` to use the new interface, but I was going to put that in a separate PR.